### PR TITLE
fix: Restarting the playout without audio send streams makes the mic indicator off.

### PR DESCRIPTION
**Pull request by cloudwebrtc**
fix: Restarting the playout without audio send streams makes the mic indicator off.

### CODE REVIEW

File: audio/audio_state.cc [hunk 1]
```diff
@@ -124,6 +124,17 @@ void AudioState::RemoveSendingStream(webrtc::AudioSendStream* stream) {
   UpdateAudioTransportWithSendingStreams();
   if (sending_streams_.empty()) {
     config_.audio_device_module->StopRecording();
+#if defined(WEBRTC_IOS)
+    auto* adm = config_.audio_device_module.get();
+    if (adm->Playing()) {
+      adm->StopPlayout();
+      if (adm->InitPlayout() == 0) {
```
Comment (davidzhao) [inline audio/audio_state.cc:131]: what does this mean? we are starting playout after stopping it?

**Review by davidzhao — APPROVED** [review]
this is a great fix! would help resolve mic indicators across our other platforms too.

mind adding a bit of comment there? both to indicate we added it but also a bit about how it works.

File: audio/audio_state.cc (hunk 1 above)
Comment (cloudwebrtc) [reply audio/audio_state.cc:131]: Through repeated tests, when the mode is changed to playback, the RTCAudioSession must be reset to stop the capture of the microphone.

Comment (cloudwebrtc) [reply audio/audio_state.cc:131]: I'll add a commen describing the intent of this PR.

Comment (davidzhao) [reply audio/audio_state.cc:131]: got it, so it'd have to be restarted after Playout is stopped?

Comment (cloudwebrtc) [reply audio/audio_state.cc:131]: If playout is stopped, we should not need to restart it, only need to deal with `adm->StopRecording()  && adm->Playing()`

It means that after the last audio send stream is closed, 1,stop recording, 2, and then reset to release the mic capture state.

**Comment by cloudwebrtc** [comment]
Replaced with a more reasonable PR. https://github.com/webrtc-sdk/webrtc/pull/52

**Pull request closed without merging**